### DISCUSSION TITLE
problems with filament installation

### DISCUSSION THREAD

**Discussion by FelipeVegaEsparza** [filamentphp/filament#8180] [Help] [answered]
Hello, I want to start working with filament 3 but when installing it gives me an error, I looked in many places and I can't find the solution.

C:\xampp\htdocs\example-app>composer require filament/filament:"^3.0-stable" -W
./composer.json has been updated
Running composer update filament/filament --with-all-dependencies
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - filament/filament 3.x-dev requires filament/support 3.x-dev -> satisfiable by filament/support[3.x-dev].
    - filament/filament v3.0.0 requires filament/support v3.0.0 -> satisfiable by filament/support[v3.0.0].
    - filament/filament v3.0.1 requires filament/support v3.0.1 -> satisfiable by filament/support[v3.0.1].
    - filament/filament v3.0.2 requires filament/support v3.0.2 -> satisfiable by filament/support[v3.0.2].
    - filament/filament v3.0.3 requires filament/support v3.0.3 -> satisfiable by filament/support[v3.0.3].
    - filament/filament v3.0.4 requires filament/support v3.0.4 -> satisfiable by filament/support[v3.0.4].
    - filament/filament v3.0.5 requires filament/support v3.0.5 -> satisfiable by filament/support[v3.0.5].
    - filament/filament v3.0.6 requires filament/support v3.0.6 -> satisfiable by filament/support[v3.0.6].
    - filament/filament v3.0.7 requires filament/support v3.0.7 -> satisfiable by filament/support[v3.0.7].
    - filament/filament v3.0.8 requires filament/support v3.0.8 -> satisfiable by filament/support[v3.0.8].
    - filament/filament v3.0.9 requires filament/support v3.0.9 -> satisfiable by filament/support[v3.0.9].
    - filament/filament v3.0.10 requires filament/support v3.0.10 -> satisfiable by filament/support[v3.0.10].
    - filament/filament v3.0.11 requires filament/support v3.0.11 -> satisfiable by filament/support[v3.0.11].
    - filament/filament v3.0.12 requires filament/support v3.0.12 -> satisfiable by filament/support[v3.0.12].
    - filament/filament v3.0.13 requires filament/support v3.0.13 -> satisfiable by filament/support[v3.0.13].
    - filament/filament v3.0.14 requires filament/support v3.0.14 -> satisfiable by filament/support[v3.0.14].
    - filament/filament v3.0.15 requires filament/support v3.0.15 -> satisfiable by filament/support[v3.0.15].
    - filament/filament v3.0.16 requires filament/support v3.0.16 -> satisfiable by filament/support[v3.0.16].
    - filament/filament v3.0.17 requires filament/support v3.0.17 -> satisfiable by filament/support[v3.0.17].
    - filament/filament v3.0.18 requires filament/support v3.0.18 -> satisfiable by filament/support[v3.0.18].
    - filament/filament v3.0.19 requires filament/support v3.0.19 -> satisfiable by filament/support[v3.0.19].
    - filament/filament v3.0.20 requires filament/support v3.0.20 -> satisfiable by filament/support[v3.0.20].
    - filament/filament v3.0.21 requires filament/support v3.0.21 -> satisfiable by filament/support[v3.0.21].
    - filament/filament v3.0.22 requires filament/support v3.0.22 -> satisfiable by filament/support[v3.0.22].
    - filament/filament v3.0.23 requires filament/support v3.0.23 -> satisfiable by filament/support[v3.0.23].
    - filament/filament v3.0.24 requires filament/support v3.0.24 -> satisfiable by filament/support[v3.0.24].
    - filament/filament v3.0.25 requires filament/support v3.0.25 -> satisfiable by filament/support[v3.0.25].
    - filament/filament v3.0.26 requires filament/support v3.0.26 -> satisfiable by filament/support[v3.0.26].
    - filament/filament v3.0.27 requires filament/support v3.0.27 -> satisfiable by filament/support[v3.0.27].
    - filament/filament v3.0.28 requires filament/support v3.0.28 -> satisfiable by filament/support[v3.0.28].
    - filament/filament v3.0.29 requires filament/support v3.0.29 -> satisfiable by filament/support[v3.0.29].
    - filament/filament v3.0.30 requires filament/support v3.0.30 -> satisfiable by filament/support[v3.0.30].
    - filament/filament v3.0.31 requires filament/support v3.0.31 -> satisfiable by filament/support[v3.0.31].
    - filament/filament v3.0.32 requires filament/support v3.0.32 -> satisfiable by filament/support[v3.0.32].
    - filament/filament v3.0.33 requires filament/support v3.0.33 -> satisfiable by filament/support[v3.0.33].
    - filament/filament v3.0.34 requires filament/support v3.0.34 -> satisfiable by filament/support[v3.0.34].
    - filament/filament v3.0.35 requires filament/support v3.0.35 -> satisfiable by filament/support[v3.0.35].
    - filament/filament v3.0.36 requires filament/support v3.0.36 -> satisfiable by filament/support[v3.0.36].
    - filament/filament v3.0.37 requires filament/support v3.0.37 -> satisfiable by filament/support[v3.0.37].
    - filament/support[v3.0.0, ..., 3.x-dev] require ext-intl * -> it is missing from your system. Install or enable PHP's intl extension.
    - Root composer.json requires filament/filament ^3.0-stable -> satisfiable by filament/filament[v3.0.0, ..., 3.x-dev].

To enable extensions, verify that they are enabled in your .ini files:
    - C:\xampp\php\php.ini
You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
Alternatively, you can run Composer with `--ignore-platform-req=ext-intl` to temporarily ignore these required extensions.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by danharrin** [comment, marked as answer]
Install the PHP intl extension as the error says

**Reply by FelipeVegaEsparza** [reply]
Thanks for the answer, I added the following line to my php.ini (extension=php_intl.dll) and it worked without problem. Now to study and learn with this tool. thank you

**Reply by edriso** [reply]
@FelipeVegaEsparza I added that line as you said and also worked for me, thanks!

**Reply by edriso** [reply]
At the time I had PHP v8.1 and Composer v2.5. I got the same error when updating PHP to v8.3 and Composer to v2.7. Uncommenting `extension=intl` or adding the line above didn't solve the error. So, currently searching for another solution.

**Reply by edriso** [reply]
I found a solution by uncomment these 2:
`extension=intl`
`extension=zip`

**Reply by fadoshi** [reply]
Here's the steps I followed that worked for me,
```
Add 'extension=php_intl.dll' in php.ini
uncomment extension=intl and extension=zip
Run 'composer require filament/filament' (without the version tag)
```
I got filament ^3.2 installed without error with these steps

**Comment by eduardonwa** [comment]
i have done all of this and it still doesn't work.

**Comment by nodesiremonk** [comment]
Same here.
Already installed and enabled `intl` and `zip` extension, but still got the same error message saying `ext-intl` is missing.
Composer version 2.7.4
PHP version 8.3.6

**Reply by eduardonwa** [reply]
Hi, for me, it turned out that the extension wasn’t installed. These were the commands i used:

```
// views the installed packages
php -i | grep ini

// intl for my version of php
sudo apt-get install php8.3-intl
```

after that i was able to use filament. No need to uncomment anything on php.ini.

**Reply by nodesiremonk** [reply]
I checked `intl` extension with `php -m` and `php -i`, as well as `phpinfo()` so I am pretty sure it has been installed.  The only difference is that I am using docker with `php:fpm-alpine` image, maybe the `intl` extension in Alpine is different from that in Debian so it has not been recognized by composer

**Reply by nodesiremonk** [reply]
Tried another Debian-based image, still the same result

**Reply by yusufrachmad** [reply]
> Hi, for me, it turned out that the extension wasn’t installed. These were the commands i used:
> 
> ```
> // views the installed packages
> php -i | grep ini
> 
> // intl for my version of php
> sudo apt-get install php8.3-intl
> ```
> 
> after that i was able to use filament. No need to uncomment anything on php.ini.

it worked like a charm. thanks!!

**Reply by raja-muhammad-asher** [reply]
> e commands i used

worked for me

**Comment by fiokee** [comment]
here is the step i followed it works add:
extension=intl
